drivers/i2c/target/eeprom_target.c

```diff
@@ -126,10 +135,19 @@ static int eeprom_target_write_received(struct i2c_target_config *config,
 		data->first_write = false;
 	} else {
 		data->buffer[data->buffer_idx++] = val;
+		data->data_modified = true;
 	}
 
 	data->buffer_idx = data->buffer_idx % data->buffer_size;
 
+	if (data->store_data && data->data_modified) {
+		if (data->store_data(data->buffer, data->buffer_size) < 0) {
+			LOG_ERR("Failed to store data");
+		}
+	}
+
+	data->data_modified = false;
```

Comment:
Should this be set only if the store data callback succeeded?

Reply:
Does not really make much difference I guess? When a new write of a register is performed, the entire array gets passed again anyway?